addons/mail/static/tests/owl/component/activity_tests.js

```diff
@@ -0,0 +1,832 @@
+                        type: 'ir.actions.act_window',
+                        res_model: 'mail.activity',
+                        res_id: 12,
+                        view_mode: 'form',
+                        views: [[false, 'form']],
+                        target: 'new',
+                    },
+                    "should open the preview mail form view"
+                );
+            },
+        },
+    });
+    const activityLocalId = this.env.store.dispatch('_createActivity', {
+        id: 12,
+        can_write: true,
+        mail_template_ids: [{ id: 1, name: "Dummy mail template" }],
+        res_id: 42,
+        res_model: 'res.partner',
+    });
+    await this.createActivityComponent(activityLocalId);
+    assert.strictEqual(
+        document.querySelectorAll('.o_Activity').length,
+        1,
+        "should have activity component in DOM"
+    );
+    const activity = document.querySelector('.o_Activity');
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_editButton').length,
+        1,
+        "should have activity mail template name preview button in DOM"
+    );
+
+    await activity.querySelector('.o_Activity_editButton').click();
+});
+
+QUnit.test('activity click on cancel', async function (assert) {
+    assert.expect(7);
+
+    await this.start({
+        async mockRPC(route, args) {
+            if (route === '/web/dataset/call_kw/mail.activity/unlink') {
+                assert.step('unlink');
+                assert.strictEqual(args.args[0].length, 1);
+                assert.strictEqual(args.args[0][0], 12);
+                return;
+            } else {
+                return this._super(...arguments);
+            }
+        },
+    });
+    const activityLocalId = this.env.store.dispatch('_createActivity', {
+        id: 12,
+        can_write: true,
+        mail_template_ids: [{ id: 1, name: "Dummy mail template" }],
+        res_id: 42,
+        res_model: 'res.partner',
+    });
+    await this.createActivityComponent(activityLocalId, {
+        parentTemplate: xml`<div>
+            <p>This text is just here to prevent template to be empty.</p>
+            <t t-if="storeProps.stillExists">
+                <Activity activityLocalId="props.activityLocalId"/>
+            </t></div>`,
+        parentUseStore: (state, props) => {
+            return { stillExists: state.activities && state.activities[props.activityLocalId] };
+        }
+    });
+    assert.strictEqual(
+        document.querySelectorAll('.o_Activity').length,
+        1,
+        "should have activity component"
+    );
+    const activity = document.querySelector('.o_Activity');
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_cancelButton').length,
+        1,
+        "should have activity cancel button"
+    );
+
+    activity.querySelector('.o_Activity_cancelButton').click();
+    await afterNextRender();
+    assert.verifySteps(['unlink'], "should have called unlink rpc");
```

Comment:
... after clicking on cancel (always give some context of assertion, it's much nicer when it fails from QUnit UI)